### PR TITLE
Add support for S3 and EBS encryption

### LINKED ISSUE #14: Add support for SE S3 Encryption and SE EBS Encryption
### Is your feature request related to a problem? Please describe.

Some environments like GovCloud require S3 and EBS encryption by default.  Need the ability to enable encryption and provide KMS keys in the AWS Cloud connector.

### Describe the solution you'd like

The module will provide two options.
1. Add inputs to provide the ARN for an S3 and EBS KMS key.  This allows users to specify pre-existing keys, or to handle the key creation outside the module.  This would be referred to as custom KMS keys.
2. Create new KMS keys for S3 and EBS encryption if none are provided, and user wants them created.

If a KMS key is provided and an "enable encryption" flag is set to true, configure encryption using the provided key.
If a KMS key is not provided, and an "enable encryption" flag is set to true, then create a new key and configure encryption with new key.
If "enable encryption" flag is set to false (default), then do not create new keys and ignore any kms keys provided.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jpeters-vmw**
Fixes #14

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -151,6 +151,26 @@ variable "se_instance_type" {
   type        = string
   default     = "c5.large"
 }
+variable "se_s3_encryption" {
+  description = "Enable encryption on SE S3 Bucket"
+  type        = bool
+  default     = "false"
+}
+variable "se_ebs_encryption" {
+  description = "Enable encryption on SE S3 Bucket"
```
Comment (slarimore02) [inline variables.tf]: looks like this needs to be updated to reflect the SE volume/ami instead

File: variables.tf [hunk 2]
```diff
@@ -151,6 +151,26 @@ variable "se_instance_type" {
   type        = string
   default     = "c5.large"
 }
+variable "se_s3_encryption" {
+  description = "Enable encryption on SE S3 Bucket"
```
Comment (slarimore02) [inline variables.tf]: could you add something about the default behavior is generate new KMS keys unless the associated key_arn variable is set to true?

**Review by slarimore02 — COMMENTED** [review]
Looks good! Would it make sense to add the kms resources to the security.tf file instead?

File: variables.tf (hunk 1 above)
Comment (jpeters-vmw) [reply variables.tf]: Good catch.  I'll fix that

**Comment by jpeters-vmw** [comment]
> Looks good! Would it make sense to add the kms resources to the security.tf file instead?

I thought about that, but security.tf seemed to be more about security groups and didn't want to confuse that so I made a new kms.tf file to handle keys.

File: variables.tf (hunk 2 above)
Comment (jpeters-vmw) [reply variables.tf]: Absolutely

**Pull request merged**